### ISSUE TITLE
Circular dependecy when importing the library

### ISSUE THREAD

**Issue by brandonfl** [fonimus/ssh-shell-spring-boot#180]
Hello, 

I tryed to import your lib but it seems to have a circular dependecy into it.
I use spring boot version 2.6.2

```
***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

   sshShellConfiguration defined in URL [jar:file:/root/.gradle/caches/modules-2/files-2.1/com.github.fonimus/ssh-shell-spring-boot-starter/1.7.0/2b5916b3bbaa08598ead00ba072cb12ec352772d/ssh-shell-spring-boot-starter-1.7.0.jar!/com/github/fonimus/ssh/shell/SshShellConfiguration.class]
      ↓
   sshShellCommandFactory defined in URL [jar:file:/root/.gradle/caches/modules-2/files-2.1/com.github.fonimus/ssh-shell-spring-boot-starter/1.7.0/2b5916b3bbaa08598ead00ba072cb12ec352772d/ssh-shell-spring-boot-starter-1.7.0.jar!/com/github/fonimus/ssh/shell/SshShellCommandFactory.class]
      ↓
   sshShell defined in class path resource [com/github/fonimus/ssh/shell/SshShellAutoConfiguration.class]
┌─────┐
|  org.springframework.shell.result.ResultHandlerConfig
└─────┘
```
Did there is something to configure and/or to exclude to avoid this ? 

Thanks in advance

**Comment by brandonfl** [comment]
From the PR from the dependabot #172 
It is also present into the test https://github.com/fonimus/ssh-shell-spring-boot/runs/4779273211

![Capture d’écran de 2022-01-17 11-20-22](https://user-images.githubusercontent.com/25805069/149751913-08747524-7cba-4607-9e89-eace2533274f.png)

**Comment by H3llK33p3r** [comment]
Same in my case with spring boot 2.6.3.

**Comment by ttimot24** [comment]
Same here

**Comment by fonimus** [comment]
Hello, thanks for posting the issue, I am aware of this issue (like you said @brandonfl it is in the current dependabot PR to bump spring boot) but unfortunately it is coming from spring-shell dependency so I cannot really do something about it immediately.
They have fixed the issue in the first milestone of next release https://github.com/spring-projects/spring-shell/releases/tag/v2.1.0-M1 so maybe the final release will be soon.

In the meantime you should be able to run app with last spring boot version and following property :
`spring.main.allow-circular-references=true` (I know it is dirty !)

**Comment by ThanasisBarekas** [comment]
Hello! 
I have searched in spring shell web page etc about a possible timeline regarding the release date of the next official release that will solve this issue but with no luck. Anyone knows more? 

Thank you,
Thanasis

**Comment by buergerj** [comment]
Spring Shell 2.1.0 has been released in the meantime. There is already a PR of dependabot:
#223 will solve this issue.

**Comment by fonimus** [comment]
Hello everyone !

Spring shell 2.1.0 has been finally released indeed, but they changed lots of stuff for a minor release..
Anyway I finally found the time to upgrade dependency, you can find it in new available release [2.0.0](https://github.com/fonimus/ssh-shell-spring-boot/releases/tag/2.0.0).

Note: I also fixed a regression (in completion part) in my code but they are aware of it and they will fix it in version 2.1.1.
